add pmi2 and pmix module, refactor openmpi-pmi2/x module, fix issue #151

dingp opened the pull request.

dingp committed:
add pmi2 and pmix module, refactor openmpi-pmi2/x module, fix issue #151

namehta4 commented:
Hi,

May I please request on the clarity for the timeline on this PR? I am planning to migrate LAMMPS container user support to podman-hpc from shifter and would be relying on these modules.
Thanks!

dingp committed:
add missing lines for pmix and pmi2 modules into setup.cfg so that they are included in built rpm

dingp commented on etc/modules.d/openmpi-pmix.yaml:
```diff
@@ -3,24 +3,7 @@ cli_arg: openmpi-pmix
 help: Enable OpenMPI/PMIx Support
 env: ENABLE_OPENMPI_PMIX
 conflicts: ["mpich", "cuda-mpi", "openmpi-pmi2"]
+depends_on: ["pmix"]
 shared_run: False
```
@copilot Is ` additional_args: ["--pmix"]` needed as well?

Copilot committed:
refactor modules: move ibverbs bind to pmi2/pmix and drop openmpi-pmi2/pmix modules

Agent-Logs-Url: https://github.com/NERSC/podman-hpc/sessions/9afbc3c4-cf8e-4803-aecd-2fa825142524

Co-authored-by: dingp <9438483+dingp@users.noreply.github.com>

Copilot committed:
revert: revert commit f3de446 to restore openmpi-pmi2/pmix modules

Agent-Logs-Url: https://github.com/NERSC/podman-hpc/sessions/0074882d-777a-43a0-bae3-3e016411f8c2

Co-authored-by: dingp <9438483+dingp@users.noreply.github.com>

dingp commented on etc/modules.d/openmpi-pmi2.yaml:
```diff
@@ -3,21 +3,7 @@ cli_arg: openmpi-pmi2
 help: Enable OpenMPI/PMI2 Support
 env: ENABLE_OPENMPI_PMI2
 conflicts: ["mpich", "cuda-mpi", "openmpi-pmix"]
+depends_on: ["pmi2"]
```
This is correct. I went back to check the initial PR which introduced the dependency and conflict support, and found that I actually left a review which remained unresolved before the merge -- that the missing dependent modules only trigger a warning, but not automatically enabled....

dingp replied:
@copilot add `additional_args: ["--pmi2"]`

Copilot committed:
feat: add additional_args to openmpi-pmi2 and openmpi-pmix modules

Agent-Logs-Url: https://github.com/NERSC/podman-hpc/sessions/e619f471-437a-4058-bf71-e38192bcc74b

Co-authored-by: dingp <9438483+dingp@users.noreply.github.com>
